xtask/src/lib.rs

```diff
@@ -79,29 +88,42 @@ fn reformat(text: impl std::fmt::Display) -> Result<String> {
 }
 
 fn ensure_rustfmt() -> Result<()> {
-    match Command::new("rustup")
-        .args(&["run", TOOLCHAIN, "--", "cargo", "fmt", "--version"])
+    match Command::new("rustfmt")
+        .args(&["--version"])
+        .env("RUSTUP_TOOLCHAIN", RUSTFMT_TOOLCHAIN)
+        .stdout(Stdio::piped())
         .stderr(Stdio::null())
-        .stdout(Stdio::null())
-        .status()
+        .spawn()
+        .and_then(|child| child.wait_with_output())
     {
-        Ok(status) if status.success() => return Ok(()),
-        _ => (),
-    };
-    run!("rustup toolchain install {}", TOOLCHAIN)?;
-    run!("rustup component add rustfmt --toolchain {}", TOOLCHAIN)?;
-    Ok(())
+        Ok(output)
+            if output.status.success()
+                && std::str::from_utf8(&output.stdout)?.contains(RUSTFMT_TOOLCHAIN) =>
```

Comment:
This doesn't check that it is the latest version. `rustup toolchain install` updates if the toolchain isn't the latest version.

Reply:
Does outdated clippy really matter? I also think it's developer's responsibility to choose the correct version of clippy.

Note that the aim of this PR is to make tests and utilities work with bare-installation of rust with components, **without rustup**.